File: TP/TP1/Code/5H.cpp
/* Afficher un damier de taille N * M */
#include <iostream>

int main()
{
  int n;
  int m;
  
  std::cout << "Quelle est la longueur de la ligne ?" << std::endl;
  std::cin >> n;
	std::cout << "Quelle est la hauteur du rectangle ?" << std::endl;
  std::cin >> m;
  
  for (int i = 0; i < m; ++i) {
    for (int j = 0; j < n; ++j) {
			if ((i % 2 == 0 && j % 2 == 0) || (i % 2 != 0 && j % 2 != 0))
        std::cout << "*" << std::flush;
      else
        std::cout << " " << std::flush;
    }
    std::cout << std::endl;
  }

  return 0;
}

File: TP/TP1/Code/Python/5H.py
#!/usr/bin/python
# Afficher un damier de taille N * M
print('Donne moi la hauteur : ', end='')
foo = int(input())
print('Donne moi la longeur : ', end='')
bar = int(input())

for i in range(foo):
    for j in range(bar):
        if (i % 2 == 0 and j % 2 == 0) or (i % 2 != 0 and j % 2 != 0):
                print('*', end='')
        else:
                print(' ', end='')
    print()  # Ici on affiche un saut de ligne
